judo-ui-react/src/main/resources/actor/src/components/widgets/AggregationInput.tsx.hbs

```diff
@@ -233,36 +227,22 @@ export const AggregationInput = ({
           <Paper>
             <ClickAwayListener onClickAway={handleDropdownClose}>
               <MenuList id={`${name}-menu`} autoFocusItem>
-                {exists(value) && onSet && !readOnly && (
-                  <MenuItem className={`${name}-set`} disabled={disabled} onClick={ (event: any) => { handleDropdownClose(event); onSet(); } }>
-                    <MdiIcon path="magnify" sx={ { mr: 2 } } />
-                    {t('judo.component.AggregationInput.open-filter', { defaultValue: 'Open selector dialog' }) as string}
-                  </MenuItem>
-                )}
-                {exists(value) && onEdit && (
-                  <MenuItem className={`${name}-edit`} disabled={disabled || editMode} onClick={ (event: any) => { handleDropdownClose(event); onEdit(); } }>
-                    <MdiIcon path="pencil" sx={ { mr: 2 } } />
-                    {t('judo.component.AggregationInput.open-editor', { defaultValue: 'Edit' }) as string}
-                  </MenuItem>
-                )}
-                {exists(value) && onDelete && (
-                  <MenuItem className={`${name}-delete`} disabled={disabled || editMode} onClick={ (event: any) => { handleDropdownClose(event); onDelete(); } }>
-                    <MdiIcon path="delete" sx={ { mr: 2 } } />
-                    {t('judo.component.AggregationInput.delete', { defaultValue: 'Delete' }) as string}
-                  </MenuItem>
-                )}
-                {exists(value) && onRemove && !readOnly && (
-                  <MenuItem className={`${name}-remove`} disabled={disabled} onClick={ (event: any) => { handleDropdownClose(event); onRemove(); } }>
-                    <MdiIcon path="link_off" sx={ { mr: 2 } } />
-                    {t('judo.component.AggregationInput.remove', { defaultValue: 'Remove' }) as string}
-                  </MenuItem>
-                )}
-                {exists(value) && onUnset && !readOnly && (
-                  <MenuItem className={`${name}-unset`} disabled={disabled} onClick={ (event: any) => { handleDropdownClose(event); onUnset(); } }>
-                    <MdiIcon path="link_off" sx={ { mr: 2 } } />
-                    {t('judo.component.AggregationInput.unset', { defaultValue: 'Unset' }) as string}
-                  </MenuItem>
-                )}
+                { aggregationInputButtonProps.filter(a => a.name === 'set' || a.name === 'delete' || a.name === 'unset').map( (aggregationInputButtonProp: AggregationInputButtonProps) => 
+                    aggregationInputButtonProp.dropDownButtonIsVisible && aggregationInputButtonProp.dropDownButtonIsVisible(value, aggregationInputButtonProp.onClick) && (
+                      <MenuItem
+                        className={aggregationInputButtonProp.className}
```

Comment:
If we are using `.map` in JSX we always need to set the `key` prop on generated nodes.

Reply:
[141d4bf](https://github.com/BlackBeltTechnology/judo-ui-react-template/pull/378/commits/141d4bf18275f59a778c628641bce06e4b946e7f)